sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/physical/partitioning.scala

```diff
@@ -72,10 +72,15 @@ case object AllTuples extends Distribution {
 /**
  * Represents data where tuples that share the same values for the `clustering`
  * [[Expression Expressions]] will be co-located in the same partition.
+ *
+ * @param requiredAllClusterKeys When true, `Partitioning` which satisfies this distribution,
+ *                               must match all `clustering` expressions in the same ordering.
  */
 case class ClusteredDistribution(
     clustering: Seq[Expression],
-    requiredNumPartitions: Option[Int] = None) extends Distribution {
+    requiredNumPartitions: Option[Int] = None,
+    requiredAllClusterKeys: Boolean = SQLConf.get.getConf(
```

Comment:
I think we should call this `requireAllClusterKeys`?

Reply:
@sunchao - sure, updated.

Reply:
+1, and if we are open to rearrange fields, it seems to be better grouping this with partition expression.

Reply:
@HeartSaVioR - I agree that would look better. I was originally putting the new field after `clustering`, but found several other places need code change. Updated now.

Reply:
Yeah we can move forward since at least two of us think it be better. :)